ceph_deploy/install.py

```diff
@@ -28,8 +28,47 @@ def sanitize_args(args):
     return args
 
 
+def detect_components(args, distro):
+    """
+    Since the package split, now there are various different ceph components to
+    install like:
+
+    * ceph
+    * ceph-mon
+    * ceph-osd
+    * ceph-mds
+
+    This helper function should parse the args that may contain specifics about
+    these flags and return the default if none are passed in (which is, install
+    everything)
+    """
+    flags = {
+        'install_osd': 'ceph-osd',
+        'install_rgw': 'ceph-radosgw',
+        'install_mds': 'ceph-mds',
+        'install_mon': 'ceph-mon',
+    }
+
+    if distro.is_rpm:
+        defaults = default_components.rpm
```

Comment:
@trhoden as you can see here, we are returning either `default_components.rpm` or `default_components.deb`

Instead of changing it in every install method I decided to just do the logic here and adapt. This should (hopefully) address your comments for all the `install.py` files that consume this list, but do let me know if you find this approach bad so we can discuss further :)